upcloud/service/load_balancer.go

```diff
@@ -96,284 +94,121 @@ func (s *Service) GetLoadBalancers(r *request.GetLoadBalancersRequest) ([]upclou
-	res, err := s.basicGetRequest(r.RequestURL())
-	if err != nil {
-		return nil, parseJSONServiceError(err)
-	}
-
-	err = json.Unmarshal(res, &backend)
-	return &backend, err
+	return &backend, s.get(ctx, r.RequestURL(), &backend)
 }
 
 // CreateLoadBalancerBackend creates a new load balancer backend.
-func (s *Service) CreateLoadBalancerBackend(r *request.CreateLoadBalancerBackendRequest) (*upcloud.LoadBalancerBackend, error) {
+func (s *Service) CreateLoadBalancerBackend(ctx context.Context, r *request.CreateLoadBalancerBackendRequest) (*upcloud.LoadBalancerBackend, error) {
 	var backend upcloud.LoadBalancerBackend
-
-	reqBody, err := json.Marshal(r)
-	if err != nil {
-		return nil, err
-	}
-
-	res, err := s.client.PerformJSONPostRequest(s.client.CreateRequestURL(r.RequestURL()), reqBody)
-	if err != nil {
-		return nil, parseJSONServiceError(err)
-	}
-
-	err = json.Unmarshal(res, &backend)
-	return &backend, err
+	return &backend, s.create(ctx, r, &backend)
 }
 
 // ModifyLoadBalancerBackend modifies an existing load balancer backend.
-func (s *Service) ModifyLoadBalancerBackend(r *request.ModifyLoadBalancerBackendRequest) (*upcloud.LoadBalancerBackend, error) {
+func (s *Service) ModifyLoadBalancerBackend(ctx context.Context, r *request.ModifyLoadBalancerBackendRequest) (*upcloud.LoadBalancerBackend, error) {
 	var backend upcloud.LoadBalancerBackend
-
-	reqBody, err := json.Marshal(r)
-	if err != nil {
-		return nil, err
-	}
-
-	res, err := s.client.PerformJSONPatchRequest(s.client.CreateRequestURL(r.RequestURL()), reqBody)
-	if err != nil {
-		return nil, parseJSONServiceError(err)
-	}
-
-	err = json.Unmarshal(res, &backend)
-	return &backend, err
+	return &backend, s.modify(ctx, r, &backend)
 }
 
 // DeleteLoadBalancerBackend deletes an existing load balancer backend.
-func (s *Service) DeleteLoadBalancerBackend(r *request.DeleteLoadBalancerBackendRequest) error {
-	if err := s.client.PerformJSONDeleteRequest(s.client.CreateRequestURL(r.RequestURL())); err != nil {
-		return parseJSONServiceError(err)
-	}
-	return nil
+func (s *Service) DeleteLoadBalancerBackend(ctx context.Context, r *request.DeleteLoadBalancerBackendRequest) error {
+	return s.delete(ctx, r)
 }
 
 // GetLoadBalancerBackendMembers retrieves a list of load balancer backend members.
-func (s *Service) GetLoadBalancerBackendMembers(r *request.GetLoadBalancerBackendMembersRequest) ([]upcloud.LoadBalancerBackendMember, error) {
+func (s *Service) GetLoadBalancerBackendMembers(ctx context.Context, r *request.GetLoadBalancerBackendMembersRequest) ([]upcloud.LoadBalancerBackendMember, error) {
 	members := make([]upcloud.LoadBalancerBackendMember, 0)
-	res, err := s.basicGetRequest(r.RequestURL())
-	if err != nil {
-		return nil, parseJSONServiceError(err)
-	}
-
-	err = json.Unmarshal(res, &members)
-	return members, err
+	return members, s.get(ctx, r.RequestURL(), &members)
 }
 
 // GetLoadBalancerBackendMember retrieves details of a load balancer balancer backend member.
```

Comment:
Just noticed `balancer balancer` typo here and in couple of other places :)

Reply:
Fixed :)